ui/src/playlist/PlaylistActions.js

```diff
@@ -23,11 +25,42 @@ import config from '../config'
 const PlaylistActions = ({ className, ids, data, record, ...rest }) => {
   const dispatch = useDispatch()
   const translate = useTranslate()
+  const dataProvider = useDataProvider()
+  const notify = useNotify()
   const isDesktop = useMediaQuery((theme) => theme.breakpoints.up('md'))
+  const { id, songCount } = record
+
+  const getAllSongs = React.useCallback(
+    (type) => {
+      dataProvider
+        .getList('playlistTrack', {
+          pagination: { page: 1, perPage: 0 },
+          sort: { field: 'id', order: 'ASC' },
+          filter: { playlist_id: id },
+        })
+        .then((res) => {
+          const data = res.data.reduce(
+            (acc, curr) => ({ ...acc, [curr.id]: curr }),
+            {}
+          )
+          const ids = Object.keys(data)
+          type === 'play'
+            ? dispatch(playTracks(data, ids))
+            : dispatch(shuffleTracks(data, ids))
+        })
+        .catch(() => {
+          notify('ra.page.error', 'warning')
+        })
+    },
+    [dataProvider, dispatch, id, notify]
+  )
 
   const handlePlay = React.useCallback(() => {
-    dispatch(playTracks(data, ids))
-  }, [dispatch, data, ids])
+    if (ids.length === songCount) {
```

Comment:
Yes, basically this, but we also need to apply the same logic to  `handlePlayNext` and `handlePlayLater`. IMO we could encapsulate this even more with a function like:
```js
getAllSongsAndDispatch(action, data, ids)
```

That could be called like:
```js
getAllSongsAndDispatch(addTracks, data, ids)
```
It would then encapsulate the `if (ids.length === songCount)` and it would not need the `type === 'play'`, ti would just need to call the `action` passed as a parameter.

Reply:
I think there is no need to pass `ids` when we fetch data directly from API, from API data come as an array and I already reduce it to an object, Passing `ids` in this case can slow the process (in case of the large playlist) because then `playTrack` and other action functions also filter the songs using `ids`. We simply want all songs in the queue so there is no need to filter them again.

Reply:
what's your opinion on this? @deluan

Reply:
Yeah, you are right, there's no need to pass the `ids` if we want to play all songs in `data`. You should remove it in that case.